src/Type/WPInputObjectType.php

```diff
@@ -112,4 +148,36 @@ public function prepare_fields( array $fields, string $type_name, array $config,
 		 */
 		return $fields;
 	}
+
+	/**
+	 * Validates type config and throws if one of type options is invalid.
+	 * Note: this method is shallow, it won't validate object fields and their arguments.
+	 *
+	 * @throws \GraphQL\Error\InvariantViolation
+	 */
+	public function assertValid(): void {
+		Utils::assertValidName( $this->name );
+
+		$fields = $this->config['fields'] ?? null;
+		if ( \is_callable( $fields ) ) {
+			$fields = $fields();
+		}
+
+		/**
+		 * @todo: This is a temporary fix to prevent the InvariantViolation from being thrown
+		 *    when the fields are not iterable. This is a temporary fix until the issue is resolved.
+		 */
+		if ( ! \is_iterable( $fields ) && ! $fields instanceof InputObjectField ) {
+			$invalidFields = Utils::printSafe( $fields );
+
+			// translators: %1$s is the name of the type and %2$s is the invalid fields
+			throw new InvariantViolation( sprintf( esc_html__( '%1$s fields must be an iterable or a callable which returns an iterable, got: %2$s.', 'wp-graphql' ), esc_html( $this->name ), esc_html( $invalidFields ) ) );
```

Comment:
The error message doesn't match the todo (which I also am not entirely sure I get, but I'm assuming we can stop this in the constructor instead of in `assertValid()`.

Reply:
@justlevine I agree that the todo doesn't make sense. I probably left it as a note to self thinking I had a different way to handle this, but clearly didn't explain well enough 🤦🏻‍♂️ 

I think we can remove the todo comment.

I don't know if we can / should move it to the constructor because of lazy loading types, so I think the assertValid is the right place as it is part of graphql-php's validation phase, which happens after types are registered but before the schema is used. This should ensure that all cross-references between types can be properly validated.

Reply:
I pushed up a change, updating the comment and error message.